lib/recurly/BaseClient.js

```diff
@@ -106,8 +113,9 @@ class BaseClient {
    */
   _errorFromResponse (resp) {
     let err = null
+    const jsonContent = (resp.contentType && resp.contentType.includes(JSON_CONTENT_TYPE))
 
-    if (resp.status < 200 || resp.status > 299) {
+    if (jsonContent && (resp.status < 200 || resp.status > 299)) {
```

Comment:
The way this conditional is written, it's not throwing an exception for non-json bodies. For instance, set the host to `v4.recurly.com`, it gives a status of `401` with non-json body. The call just returns a null instead of throwing an exception. Here's a diff that gets it to work:

```diff
diff --git a/lib/recurly/BaseClient.js b/lib/recurly/BaseClient.js
index aac55be..fcf1c00 100644
--- a/lib/recurly/BaseClient.js
+++ b/lib/recurly/BaseClient.js
@@ -115,25 +115,27 @@ class BaseClient {
     let err = null
     const jsonContent = (resp.contentType && resp.contentType.includes(JSON_CONTENT_TYPE))
 
-    if (jsonContent && (resp.status < 200 || resp.status > 299)) {
-      const errBody = resp.body && JSON.parse(resp.body).error
-      // If we have a body, we determine the error based on
-      // the contents of the body
-      if (errBody) {
-        let className = utils.classify(errBody.type)
-        if (!className.endsWith('Error')) className += 'Error'
-        const ErrClass = apiErrors[className] || ApiError
-        if (ErrClass === apiErrors.TransactionError) {
-          err = new ErrClass(errBody.message, errBody.type, {
-            transactionError: resources.TransactionError.cast(errBody.transaction_error)
-          })
-        } else {
-          err = new ErrClass(errBody.message, errBody.type, {
-            params: errBody.params || []
-          })
+    if (resp.status < 200 || resp.status > 299) {
+      if (jsonContent) {
+        const errBody = resp.body && JSON.parse(resp.body).error
+        // If we have a body, we determine the error based on
+        // the contents of the body
+        if (errBody) {
+          let className = utils.classify(errBody.type)
+          if (!className.endsWith('Error')) className += 'Error'
+          const ErrClass = apiErrors[className] || ApiError
+          if (ErrClass === apiErrors.TransactionError) {
+            err = new ErrClass(errBody.message, errBody.type, {
+              transactionError: resources.TransactionError.cast(errBody.transaction_error)
+            })
+          } else {
+            err = new ErrClass(errBody.message, errBody.type, {
+              params: errBody.params || []
+            })
+          }
+          err._setResponse(resp)
+          return err
         }
-        err._setResponse(resp)
-        return err
       // if we don't have a JSON body, we determine the error
       // based on the http status code
       } else {
```

This is getting a bit complex but I have a plan to refactor it if you don't see any obvious ways to make it less branchy.